packages/pg/lib/connection.js

```diff
@@ -152,6 +152,44 @@ class Connection extends EventEmitter {
     this._send(serialize.query(text))
   }
 
+  // query using single batch of packets
+  // https://github.com/brianc/node-postgres/issues/3325
+  queryWithPacketBatching(query, valueMapper) {
+    const packets = []
+
+    if (!query.hasBeenParsed(this)) {
+      packets.push(serialize.parse({
+        text: query.text,
+        name: query.name,
+        types: query.types,
+      }))
+    }
+
+    packets.push(serialize.bind({
+      portal: query.portal,
+      statement: query.name,
+      values: query.values,
+      binary: query.binary,
+      valueMapper: valueMapper,
+    }))
```

Comment:
This seems to have lost the `try`? I’m not sure if it’s a bug in #1855’s test or if the rewrite to pg-protocol or some other change made the `try` unnecessary, but it should probably be consistent with `prepare`.